src/mongo-service.js

```diff
@@ -198,35 +201,33 @@ class MongoService extends MongoQueryService {
       return validated;
     }));
 
-    const updatePromises = updated.map((doc) => this._collection.update(
-      { ...query, _id: doc._id },
-      doc,
-      { ...options, replaceOne: true },
-    ));
-
-    await Promise.all(updatePromises);
-
-    updated.forEach((doc, index) => {
+    const updateCallback = (doc) => {
       this._bus.emit('updated', {
         doc,
-        prevDoc: docs[index],
+        prevDoc: docs.find((d) => d._id === doc._id),
       });
-    });
+    };
+
+    await this._collection.updateMany(
```

Comment:
the same, does it replace document instead set?

Reply:
probably need to specify updateFn inside